ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml

```diff
@@ -46,6 +46,40 @@ tests:
 - as: e2e-aws
   steps:
     cluster_profile: aws
+    post:
+    - as: cleanup-infra
+      cli: latest
+      commands: |-
+        bin/hypershift destroy infra aws \
+          --infra-id $(cat "${SHARED_DIR}/infra-id")
+          --aws-creds "${CLUSTER_PROFILE_DIR}/.awscred" && \
```

Comment:
Should we run `destroy iam` regardless of the outcome of infra? We only get one chance to execute